packages/server/src/documents/documents.controller.ts

```diff
@@ -91,6 +76,58 @@ export class DocumentsController {
     });
   }
 
+  async saveDoc(document: Document, user: User) {
+    const createResult: Document = await this.centrifugeService.documents.createDocumentV2(
+      user.account,
+      {
+        attributes: document.attributes,
+        read_access: document.header.read_access
+          ? document.header.read_access
+          : [],
+        write_access: document.header.write_access
+          ? document.header.write_access
+          : [],
+        scheme: CoreapiCreateDocumentRequest.SchemeEnum.Generic,
+      },
+    );
+
+    return  await this.databaseService.documents.insert({
+      ...createResult,
+      attributes: unflatten(createResult.attributes),
```

Comment:
Do we still need the flatten/unflatten functions. This was done because we needed to have flat object on the node. Ex:'funding_agreement.id', 'funding_agreement.signature' etc. As I remember we were flattening before sending the node and unflattening on the gateway

Reply:
we are currently not using funding agreements in the use case but we still haven't removed funding agreements code from the node so i would still leave this to keep it in sync with the way node would expect attributes for now